Justfile

```diff
@@ -1,28 +1,35 @@
 set shell := ["bash", "-uc"]
 
-build:
+# Setup local development environment
+setup:
+  #!/bin/bash
+  if [[ "$(cargo 2>&1)" == *"rustup could not choose a version of cargo to run"* ]]; then
+    rustup default stable
+  fi
+
+build: setup
   cargo build --release
 
-test:
+test: setup
```

Comment:
adding ` setup` for in these lines runs `just setup` first?

Reply:
that's right

we're using `rustup` which is an aggregation of many underlying rust dependencies, but it has to be setup (`rustup default stable` downloads and installs all the latest stable versions). by adding `setup` as a dependency, that step is automatically handled.

cc @nitro-neal same question